tests/src/test/scala/cats/tests/TrySuite.scala

```diff
@@ -61,6 +61,17 @@ class TrySuite extends CatsSuite {
       res should not be (null)
     }
   }
+
+  test("catchOnly works") {
+    forAll { e: Either[String, Int] =>
+      val str = e.fold(identity, _.toString)
+      val res = MonadError[Try, Throwable].catchOnly[NumberFormatException](str.toInt)
+      // the above should just never cause an uncaught exception
+      // this is a somewhat bogus test:
+      res should not be (null)
+    }
+  }
```

Comment:
Might it be useful to also test that `catchOnly` on an unrelated exception does not catch?